go/src/koding/kites/kloud/provider/google/google.go

```diff
@@ -84,6 +85,12 @@ func (c *Cred) Valid() error {
 		return errors.New(`cred value for "project" is empty`)
 	}
 
+	var raw json.RawMessage
+
+	if err := json.Unmarshal([]byte(c.Credentials), &raw); err != nil {
```

Comment:
/cc @ppknap 

Best effort method, ensuring the Credentials are a valid JSON, should filter out attempts of inputing random text. If you want more (e.g. validating the actual schema), please commit it to the branch of this PR.